ci/test-conformance-gke.sh

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -exu
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+GKE_ZONE="us-west1"
+GKE_HOST="UBUNTU"
+GKE_SERVICE_CIDR="10.94.0.0/16"
+GKE_PROJECT="antrea"
+
+RUN_ALL=true
+RUN_SETUP_ONLY=false
+RUN_CLEANUP_ONLY=false
+TEST_FAILURE=false
+
+_usage="Usage: $0 [--cluster-name <GKEClusterNameToUse>] [--k8s-version <ClusterVersion>] \
+                  [--svc-account <Name>] [--svc-account-key <Key>]  \
+                  [--gke-project <Project>] [--gke-zone <Zone>] [--svc-cidr <ServiceCIDR>] [--host-type <HostType] \
+                  [--setup-only] [--cleanup-only]"
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --cluster-name)
+    CLUSTER="$2"
+    shift 2
+    ;;
+    --svc-account-key)
+    SVC_ACCOUNT_KEY="$2"
+    shift 2
+    ;;
+    --svc-account)
+    SVC_ACCOUNT_NAME="$2"
+    shift 2
+    ;;
+    --gke-project)
+    GKE_PROJECT="$2"
+    shift 2
+    ;;
+    --gke-zone)
+    GKE_ZONE="$2"
+    shift 2
+    ;;
+    --svc-cidr)
+    GKE_SERVICE_CIDR="$2"
+    shift 2
+    ;;
+    --host-type)
+    GKE_HOST="$2"
+    shift 2
+    ;;
+    --k8s-version)
+    K8S_VERSION="$2"
+    shift 2
+    ;;
+    --setup-only)
+    RUN_SETUP_ONLY=true
+    RUN_ALL=false
+    shift
+    ;;
+    --cleanup-only)
+    RUN_CLEANUP_ONLY=true
+    RUN_ALL=false
+    shift
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+function setup_gke() {
+
+    if [[ -z ${CLUSTER+x} ]]; then
+        CLUSTER="${JOB_NAME}-${BUILD_NUMBER}"
+    fi
+    if [[ -z ${K8S_VERSION+x} ]]; then
+        K8S_VERSION="1.15.11-gke.11"
+    fi
+
+    echo "=== This cluster to be created is named: ${CLUSTER} ==="
+    echo "CLUSTERNAME=${CLUSTER}" > ci_properties.txt
+
+    echo "=== Using the following awscli version ==="
+    gcloud --version
+
+    # gcloud auth login requires verification code from url
+    gcloud auth activate-service-account  --key-file=${SVC_ACCOUNT_KEY}
+    echo "=== Using the following kubectl ==="
+    which kubectl
+
+    echo '=== Creating a cluster in GKE ==='
+    gcloud container --project ${GKE_PROJECT} clusters create ${CLUSTER} --image-type ${GKE_HOST} \
```

Comment:
the project needs to be created ahead of time?

if yes maybe something worth mentioning in the `_usage` help string

Reply:
Done